webapp/tests/karma/unit/controllers/reports.js

```diff
@@ -2,11 +2,18 @@ describe('ReportsCtrl controller', () => {
 
   'use strict';
 
+  const nestedPromiseKarmaGarbage = promise => Promise.resolve()
+    .then(() => Promise.resolve())
+    .then(() => Promise.resolve())
+    .then(promise);
```

Comment:
This should not be necessary. In other tests we replace `$q` with `Q` which solves this issue, eg:

```js
return controller('ContactsContentCtrl', {
  '$q': Q,
  ...
```